### ISSUE TITLE
New-ScheduledTaskSettingsSet: add `-Delay` option from `schtasks`

### ISSUE THREAD

**Issue by soredake** [PowerShell/PowerShell#23978]
### Summary of the new feature / enhancement

`schtasks` has `/DELAY` option that allows you to set delay of task startup, unfortunately, `New-ScheduledTaskSettingsSet` does not have such option, would be nice if it can be added.

https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/schtasks-create

**Comment by mklement0** [comment]
Note that the [`ScheduledTasks`](https://learn.microsoft.com/en-us/powershell/module/scheduledtasks) module that `New-ScheduledTaskSettingsSet` is a part of isn't being maintained in this repo, so the standard advice applies (taken from the now _obsolete_ [PowerShellModuleCoverage](https://github.com/PowerShell/PowerShellModuleCoverage) repo, emphasis added):

> NOTE To **report problems with _modules that ship in Windows_, use the _Windows Feedback Hub_**. For more information, see [Send feedback to Microsoft with the Feedback Hub app](https://support.microsoft.com/windows/send-feedback-to-microsoft-with-the-feedback-hub-app-f59187f8-8739-22d6-ba93-f66612949332).

**Comment by jborean93** [comment]
The delay option is defined as part of a trigger and not the whole task settings. It's specifically only valid for a trigger set for on startup, logon, or events. While the [New-ScheduledTaskTrigger](https://learn.microsoft.com/en-us/powershell/module/scheduledtasks/new-scheduledtasktrigger?view=windowsserver2022-ps) cmdlet has a `-RandomDelay` parameter this isn't the same as a delay. Luckily the output object from `New-ScheduledTaskTrigger` can be changed to add your delay (the format is in the ISO 8601 duration format)

```powershell
$trigger = New-ScheduledTaskTrigger -AtStartup
$trigger.Delay = "PT5M" # 5 minutes
```

**Comment by mklement0** [comment]
@jborean93:

* _Fundamentally_, this issue doesn't belong in this repo, as argued in my previous comment.

* _Incidentally_, you - commendably - provided feedback that presumably renders raising the issue at hand unnecessary (wherever it may appropriate).

While _in combination_ our comments provide all necessary information and - thanks to you - then some, the fact that you chose to post your comment _in isolation_, without acknowledging the _fundamental_ point may sow confusion:

That is, this isolated juxtaposition of our comments may cause them to be mis-perceived as _competing with_ / _contradicting_ each other, even though they do not (right?).

When providing _incidental_ advice in the future, I therefore suggest also acknowledging the fundamental issue of whether a given issue belongs in this repo or not - either by mentioning it yourself, or by acknowledging a previous comment to that effect.

**Comment by mklement0** [comment]
@jborean93, I do appreciate your thoughtful response, and, with the utmost sincerity, let me return the favor:

👎

I trust that this fully conveys my thoughts on the matter.

----

On the off chance that it doesn't, however:

* I genuinely appreciated your insight on the issue at hand, as I've tried to express in my previous comment.

* Separately - and more importantly - I do think it is important to provide _consistent guidance_ to posters as to whether their issues are _appropriate for this repo_ (something the new-issue template could provide better guidance on, but that is a separate discussion).

* As such, I think that respondent A saying "this isn't the right repo" and respondent B saying "here's a solution, which  just so happens to obviate the need for raising your issue" without reinforcing the more salient point that this is the _wrong repo_ is to the detriment of the community at large.

**Comment by mklement0** [comment]
Let's try to fix the problem at the source:

* #23981

**Comment by MartinGC94** [comment]
@mklement0 You are right that the issue shouldn't have been posted here in the first place and ideally he would have said so in his answer. However, I don't think there's any need to lecture him just because his answer wasn't 100% up to your standards. Maybe that's not how you meant it, but that's how I perceive it, and judging by the 👎 response I'm guessing he thought something similar.

**Comment by mklement0** [comment]
I appreciate the feedback, @MartinGC94, and I'm hoping that implementing #23981 will make this a moot point in the future.

The intent behind my comment definitely wasn't to lecture, and it also wasn't about any notion of standard per se.

Rather, it was about suggesting a _shared approach to curating off-topic issues_, in the hopes of reducing their incidence.
I went out of my way to point out that the specific response was helpful, but that it is also important to convey the bigger picture in a consistent, unambiguous manner.

**Comment by mklement0** [comment]
Mulling this over some more, @MartinGC94, I'm now asking for feedback from you:

* What about my initial comment, specifically, caused you to perceive it as my trying to _lecture_?

* Did you notice my attempt to acknowledge what was _good_ about the response in question? If not, what may have contributed to your not seeing that?

* Is suggesting a behavioral change to align with a - presumed - shared goal in and of itself lecturing? (I don't think so.)

In short:

* Assuming you agree with what I tried to express in my original comment: How could it (reasonably) be framed in a way that prevents misconstruing it as lecturing?

**Comment by MartinGC94** [comment]
IMO there was just no need for an additional comment after his response. You had already said this wasn't the right place and his answer didn't directly contradict that. I haven't seen enough "bad" posts like this where I feel like the community has to change its approach on how to deal with them so there's no need to suggest behavioral changes.

If I felt like it was a problem and I felt like I had to clarify it to OP and nudge a behavioral change I'd probably be less direct and say something like:
> Great answer from `<@Person>` but as stated before, Windows PowerShell modules are handled by separate teams so any issues with them should be reported in Feedback Hub.

**Comment by mklement0** [comment]
Thanks, @MartinGC94  - I truly appreciate your continued feedback.

> and his answer didn't directly contradict that.

Indeed, it did not, but for the reasons previously stated and reiterated below I feel that isn't enough. 

> where I feel like the community has to change its approach

I've personally come across so many cases of off-topic reports that I've crafted _canned responses_ that I can reuse, which is what motivates my concern that _consistent guidance_ is necessary.

And I - perhaps naively - assumed that a high-level contributor would see the same necessity once it was pointed out to them.

Again, as I've since realized and fleshed out, the proper solution is to incorporate such guidance into the issue templates, in the context of which an official help topic would need to be crafted, which comments on future off-topic issues can simply _link to_ (with the primary hope being that the added template information will _prevent_ creation of most off-topic  issues to begin with).

However, leaving the specific concern aside, your suggestion is worth debating on a general note:

>  I had to clarify it to OP and nudge a behavioral change

Your suggestion of talking _past_ the other commenter strikes me as suboptimal for two reasons:

* It would rely on the OP needing to believe _me_, specifically.

* I'm not sure that this indirect phrasing would result in the other commenter picking up on my desire to - in essence -  induce them to _form a new habit_.

  * If the suggested habit _doesn't make sense to them_, that's an opportunity to _talk it out_ so that a _shared understanding_ can be reached - an opportunity that is missed by a mere 👎

  * I suppose it's a "damned if you do and damned if you don't" situation: I get to choose between being _indirect_ at the risk of the message being _missed_ and being _direct_ at the risk of the message being _ignored_ due to personality issues.

**Comment by MartinGC94** [comment]
> I've personally come across so many cases of off-topic reports that I've crafted canned responses that I can reuse, which is what motivates my concern that consistent guidance is necessary.

I agree that there's plenty of off-topic reports, but do people generally answer those reports without any indication that they are wrong? That hasn't been my experience but it's possible I've simply missed them as I rarely look at closed issues that I haven't responded to.

> It would rely on the OP needing to believe me, specifically.

But that's always going to be the case, isn't it? I mean regardless if we take the direct or indirect approach you need the helpful helper to acknowledge your point with a 👍 or normal comment. I think that's more likely to succeed if you write a short and simple comment that is easily agreeable with.

> I'm not sure that this indirect phrasing would result in the other commenter picking up on my desire to - in essence - induce them to form a new habit.

Sure, it's not perfect because people can miss the entire point. I'm just not convinced that the harsher direct response leads to a positive outcome more often than the softer indirect response. But like you implied, neither approach is perfect you just need to try and pick the right one for each situation.

**Comment by mklement0** [comment]
Again, I appreciate the thoughtful feedback; I considered concluding this exchange with the 👍 on your most recent comment, but let me add a few more thoughts:

> but do people generally answer those reports ...

I didn't spell it out, but it isn't about _people in general_; my comment was addressed at a specific, _talented and high-profile contributor_, who - again, perhaps naively - I presumed to have a shared interest in curating this repository _collaboratively_.

> you need the helpful helper to acknowledge your point with a 👍 or normal comment.

I'm not sure that would happen without agreed-upon collaboration (and indeed it didn't happen here, on the contrary), and a _unilateral_ attempt to make an authoritative proclamation could - justifiably - be met with suspicion / be perceived as presumptuous.

And, FWIW, even though I'm taking in the feedback re _harsher_, I'm still personally personally disappointed that my well-meaning recommendation for a shared modus operandi going forward was met the way it was.
